### PR TITLE
NEW: Add dart-sdk build recipe

### PR DESCRIPTION
**Pull request by carterbox**
Checklist

- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [ ] Package does not vend other packages
- [x] Build number is 0
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your
      recipe (see
      [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos)
      for more details)
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

File: recipes/dart/build.sh [hunk 1]
```diff
@@ -0,0 +1,3 @@
+cp -R bin $PREFIX
```
Comment (scopatz) [inline recipes/dart/build.sh]: Please add `set -e`

Comment (scopatz) [reply recipes/dart/build.sh]: Also is there any way to actually build dart, rather than relying on pre-built binaries?

Comment (carterbox) [reply recipes/dart/build.sh]: Yes. It is possible to build the Dart SDK from source, but the pre-built version probably costs me less time and uses less conda-forge resources. Additionally I don't have a macOS system to test on. You're free to insist that this Dart SDK is built using the conda-forge compilers, however.

Comment (scopatz) [reply recipes/dart/build.sh]: Yep the default is to build on conda-forge.  We only repackage binaries in exceptional circumstances

Comment (scopatz) [reply recipes/dart/build.sh]: No need to worry about CI time resources

Comment (carterbox) [reply recipes/dart/build.sh]: Aight. I'll be back in a month with a recipe that builds dart from source.

**Pull request closed without merging**